### PR TITLE
feat: enforce CVE key format for releaseNotes

### PR DESCRIPTION
**Pull request by seanconroy2021**
## Describe your changes
* add regex pattern for releaseNotes.cves[].key in dataKeys schema
* move old test data CVEs from content.images.cves to releaseNotes.cves.
* update check-data-keys tests to read schema from configmap.  We can now do local testing and don't need to create two PRs.
* update mock.sh and pre-apply-task-hook.sh to load local schema via configmap
* add a test that fails when CVE key does not match CVE-<4>-<4+>

Slack: https://redhat-internal.slack.com//archives/C04PZ7H0VA8/p1763033457076779thread_ts=1762979086.594839&cid=C04PZ7H0VA8
## Checklist before requesting a review
- [ ] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [ ] My commit message includes `Signed-off-by: My name <email>`
- [ ] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)
- [ ] I have run the README.md generator script in `.github/scripts/readme_generator.sh` and verified the results using `.github/scripts/check_readme.sh`

### CODE REVIEW

**Comment by seanconroy2021** [comment]
`/var/workdir/release/9bc4c279-84f2-4395-a8d0-86c93fa082e9/data.json::$.releaseNotes.cves[0].key: 'https://example.com/security/cve/cve-2025-1234' does not match '^CVE-[0-9]{4}-[0-9]{4,}$'`

**Comment by seanconroy2021** [comment]
I had never thought of using a ConfigMap for this 😅

So we now  create a  ConfigMap from dataKeys.json before the PipelineRun, and the mock.sh curl just uses kubectl and outputs it to /tmp/schema. That means we can run the task locally against the real schema and don’t need to open two separate PRs anymore.


Example of error if CVE is malformed: 
`/var/workdir/release/9bc4c279-84f2-4395-a8d0-86c93fa082e9/data.json::$.releaseNotes.cves[0].key: 'https://example.com/security/cve/cve-2025-1234' does not match '^CVE-[0-9]{4}-[0-9]{4,}$'`

File: schema/dataKeys.json [hunk 1]
```diff
@@ -205,7 +205,8 @@
             "properties": {
               "key": {
                 "type": "string",
-                "description": "The key of the CVE e.g. CVE-3414"
+                "description": "The key of the CVE (e.g. CVE-2025-1234); must not be a URL",
+                "pattern": "^CVE-[0-9]{4}-[0-9]{4,}$"
```
Comment (johnbieren) [inline schema/dataKeys.json:209]: I am not a regex expert, so idk what `{4,}` means, but it can be five digits at the end. Like CVE-2025-12345 is valid (idk if it can go up to six digits.. @p-rog might)

Comment (seanconroy2021) [reply schema/dataKeys.json:209]: {4} means exactly 4 digits (year part).
{4,} means at least 4 digits (4 or more for the id).

**Review by happybhati — DISMISSED** [review]

File: tasks/managed/check-data-keys/tests/test-check-data-keys.yaml [hunk 2]
```diff
@@ -108,19 +108,16 @@ spec:
...
-                            }
-                          }
-                        }
+                        "purl": "pkg:example/openstack@sha256:abcde?repository_url=quay.io/example/rhosp16-rhel8"
                       }
                     ]
                   },
+                  "cves": [
+                    {
+                      "key": "CVE-2025-12345",
```
Comment (elenagerman) [inline tasks/managed/check-data-keys/tests/test-check-data-keys.yaml:97]: lines 70, 74 & 82: shouldn't it be updated as well to use properly formatted CVE IDs that match the regex pattern

File: tasks/managed/check-data-keys/tests/test-check-data-keys-strip-git-suffix.yaml [hunk 3]
```diff
@@ -49,7 +49,7 @@ spec:
               value: "$(params.trustedArtifactsDebug)"
         steps:
           - name: setup
-            image: quay.io/konflux-ci/release-service-utils:02a8ddcd16113371a255fd1ef0a196399d300162
+            image: quay.io/konflux-ci/release-service-utils:8ba7aa37b68e7a647e27e2e2456c26d29aa85b45
```
Comment (elenagerman) [inline tasks/managed/check-data-keys/tests/test-check-data-keys-strip-git-suffix.yaml:52]: lines 71, 75 & 83: shouldn't it be updated as well to use properly formatted CVE IDs that match the regex pattern

**Comment by elenagerman** [comment]
you can consider adding two more test cases (negative)
    - a 3-digit ID to ensure it fails (e.g., CVE-2025-123)
    - lowercase cve- to ensure it fails

**Comment by elenagerman** [comment]
maybe it's worth updating the task README to mention CVE format requirements?

File: tasks/managed/check-data-keys/tests/test-check-data-keys-strip-git-suffix.yaml (hunk 3 above)
Comment (seanconroy2021) [reply tasks/managed/check-data-keys/tests/test-check-data-keys-strip-git-suffix.yaml:52]: Fixed TY

File: tasks/managed/check-data-keys/tests/test-check-data-keys.yaml (hunk 2 above)
Comment (seanconroy2021) [reply tasks/managed/check-data-keys/tests/test-check-data-keys.yaml:97]: I had the same object in two places removed the top one. TY

**Comment by seanconroy2021** [comment]
> you can consider adding two more test cases (negative) - a 3-digit ID to ensure it fails (e.g., CVE-2025-123) - lowercase cve- to ensure it fails

I have added them to check-data-keys-fail-malformed-cve-key test

**Comment by seanconroy2021** [comment]
> maybe it's worth updating the task README to mention CVE format requirements?


We haven't updated the README to include any other required patterns or keys (I think that would get messy and out of sync very quickly). The idea to me is that check-data-keys is just the JSON validator; the logic is actually in the schema.

Looking at the README, it needs to be updated to remove as it outdated
`Currently, releaseNotes and CDN are the only supported systems.`
to
 `The validation schema is defined in schema/dataKeys.json in this repository.`

**Review by elenagerman — APPROVED** [review]

**Review by happybhati — APPROVED** [review]
